woocommerce-services.php

```diff
@@ -528,6 +528,175 @@ public function init_core_wizard_payments_config() {
+					wc_gateway_ppec()->get_admin_setting_link()
+				);
+				$api_creds_template = __( 'Payments will be authenticated by WooCommerce Services and directed to the following email address. To disable this and link an account, <a href="%s">click here</a>.', 'woocommerce-services' );
+
+				$api_creds_text = sprintf( $api_creds_template, add_query_arg( 'environment', 'live', $reset_link ) );
+				$settings_meta['api_credentials']['description'] = $api_creds_text;
+
+				$sandbox_api_creds_text = sprintf( $api_creds_template, add_query_arg( 'environment', 'sandbox', $reset_link ) );
+				$settings_meta['sandbox_api_credentials']['description'] = $sandbox_api_creds_text;
+
+				unset( $settings_meta['api_username'], $settings_meta['api_password'], $settings_meta['api_signature'], $settings_meta['api_certificate'] );
+				unset( $settings_meta['sandbox_api_username'], $settings_meta['sandbox_api_password'], $settings_meta['sandbox_api_signature'], $settings_meta['sandbox_api_certificate'] );
+
+			} else {
+				$reset_link = add_query_arg(
+					array( 'reroute_requests' => 'yes', 'nonce' => wp_create_nonce( 'reroute_requests' ) ),
+					wc_gateway_ppec()->get_admin_setting_link()
+				);
+				$api_creds_template = __( 'To authenticate payments with WooCommerce Services, <a href="%s">click here</a>.', 'woocommerce-services' );
+
+				if ( empty( $settings['api_username'] ) ) {
+					$api_creds_text = sprintf( $api_creds_template, add_query_arg( 'environment', 'live', $reset_link ) );
+					$settings_meta['api_credentials']['description'] .= '<br /><br />' . $api_creds_text;
+				}
+				if ( empty( $settings['sandbox_api_username'] ) ) {
+					$api_creds_text = sprintf( $api_creds_template, add_query_arg( 'environment', 'sandbox', $reset_link ) );
+					$settings_meta['sandbox_api_credentials']['description'] .= '<br /><br />' . $api_creds_text;
+				}
+			}
+
+			return $settings_meta;
+		}
+
+		public function paypal_ec_maybe_set_reroute_requests() {
+			if (
+				empty( $_GET['reroute_requests'] ) ||
+				empty( $_GET['nonce'] ) || ! wp_verify_nonce( $_GET['nonce'], 'reroute_requests' )
+			) {
+				return;
+			}
+
+			$ppec_settings = get_option( 'woocommerce_ppec_paypal_settings', array() );
+			$ppec_settings['reroute_requests'] = 'yes' === $_GET['reroute_requests'] ? 'yes' : 'no';
+			if ( isset( $_GET['environment'] ) ) {
+				$ppec_settings['environment'] = 'sandbox' === $_GET['environment'] ? 'sandbox' : 'live';
+			}
+			update_option( 'woocommerce_ppec_paypal_settings', $ppec_settings );
+
+			wp_safe_redirect( wc_gateway_ppec()->get_admin_setting_link() );
+		}
+
+		public function paypal_ec_settings( $settings ) {
+			$settings['paymentaction'] = 'sale';
+
+			if ( empty( $settings['api_username'] ) ) {
+				$settings['api_username'] = '';
+			}
+			if ( empty( $settings['sandbox_api_username'] ) ) {
+				$settings['sandbox_api_username'] = '';
+			}
```

Comment:
Can you explain these `empty()` checks and empty strings?

Reply:
The PPEC plugin shows the "prompt to connect" if the option set to `''` but not if it's unset, so I was trying to fill in the default here without having to save. I've since realized we need to do this for other options too to make it work out of the box, so I'm looking for a different way to just initialize the settings to their defaults.